### DISCUSSION TITLE
Json de l'appel API d'un service externe et pièces jointes

### DISCUSSION THREAD

**Discussion by IhsaneRQ** [MTESSDev/FRW#86] [Annonces]
Bonjour,

J'ai vu l'exemple suivant, du Json qu'on va recevoir du MESS une fois le formulaire est transmit : https://raw.githubusercontent.com/MTESSDev/FRW/main/Documentation/exemple-json-appeler-service-web.md

On veut stocker ce Json dans notre base de données, avez-vous une idée combien de caractères peut contenir ce fichier pour décider quel type de données on met dans la table SQL?
J'aimerai savoir qu'est ce que contient la balise "fichier"?
Aussi, j'aimerais savoir si vous avez un exemple des pièces jointes?

Merci d'avance et bonne journée

**Comment by anisite** [comment]
Bonjour,

Le détail de tous les champs de l'exemple ci-dessus est disponible dans la documentation github ici : https://github.com/MTESSDev/FRW/blob/main/Documentation/fichiers-transmission.md#variables-et-contenu

La taille est biensûr très variable parce que dans notre exemple il y a des fichiers "documentsProduits" grosso-modo, tout ce qui est produit par une des tâches `genererPdf` ou `genererWord` est transmis _**en même temps que le call API des données de `appelerServiceExterne`**_. 

Supposons que vous avez 3 gabarits pdf à faire remplir par FRW, ils ont chacun 1 mo il faut compter 20 % plus pour une string en base64 (binaire du PDF produit en sortie dans le json qu'on vous transmet) la taille sera donc environ 3,6 mo. Plus les données.

Il est possible pour vous de stocker ces fichiers là dans un répertoire et de conserver uniquement les données dans votre BD, au besoin. 

Nous avons une colonne "json" de type varchar(max) pour la plupart de nos json mais nous y stockons jamais de fichiers en base64 à ma connaissance.

la "balise" fichier est dans l'objet `documentsProduits`, selon la doc. il s'agit de `Le contenu du fichier en base64`.

Dans le json de l'exemple, il y a une pièce jointe, regardez l'élément `documentsSoumis`, il contient la liste des documents soumis par les utilisateurs (pièces jointes au formulaire). C'est la tâche traiterDocumentsSoumis qui renseigne cet objet en préparant et validant vos fichiers.

Nous revenir s'il y a quoi que ce soit. Merci

**Reply by IhsaneRQ** [reply]
Merci beaucoup et bonne journée
